### PR TITLE
Ens hook unittest

### PR DESCRIPTION
**Pull request by schmanu**
I checked out how exactly the gnosis SafeProvider gets the safe info when to be able to write Unit Tests for our components / hooks.

This PR provides 100% test coverage for our ens hook as a proof of concept. 

I refactored the SafeInfo Mock Code into a new test util file.

This enables us to unit test all our hooks. Its a bit ugly because we have to write test components which use our hooks but AFAIK its the only way to do this.

### CODE REVIEW

File: src/hooks/ens.ts [hunk 1]
```diff
@@ -31,15 +31,15 @@ export interface EnsResolver {
...
   const resolveCache = useMemo(() => new Map<string, string | null>(), []);
 
   const lookupCache = useMemo(() => new Map<string, string | null>(), []);
 
   const cachedResolveName = useCallback(
     async (ensName: string) => {
       const cachedAddress = resolveCache.get(ensName);
-      const resolvedAddress = cachedAddress ? cachedAddress : await web3Provider.resolveName(ensName);
+      const resolvedAddress =
+        typeof cachedAddress !== "undefined" ? cachedAddress : await web3Provider.resolveName(ensName);
```
Comment (bh2smith) [inline src/hooks/ens.ts:42]: Looks like introducing the tests revealed a potential bug here... or was this just an adaption to accommodate the test itself?

File: src/test/util.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { SafeInfo } from "@gnosis.pm/safe-apps-sdk";
 import { TokenInfo } from "../utils";
 
 const dummySafeInfo: SafeInfo = {
-  safeAddress: "0x123",
+  safeAddress: "0x1230000000000000000000000000000000000000",
```
Comment (bh2smith) [inline src/test/util.ts:6]: yea, I had a feeling this wouldn't last like this. Glad the test picked it up.

**Review by bh2smith — APPROVED** [review]
I think this is very nice! Looks like a lot of scaffolding, but I presume - now that its here - some of it can be reused for future tests.

Very nice clean code introduced here! 🎸

File: src/hooks/ens.ts (hunk 1 above)
Comment (schmanu) [reply src/hooks/ens.ts:42]: null was not cached. So names which can't be resolved where retried and not cached.

**Pull request merged**